aiohttp/web_ws.py

```diff
@@ -462,6 +453,15 @@ async def close(
             self._set_code_close_transport(WSCloseCode.ABNORMAL_CLOSURE)
             return True
 
+        # we need to break `receive()` cycle before we can call
+        # `reader.read()` as `close()` may be called from different task
+        if self._waiting:
+            if not self._close_wait:
```

Comment:
now we have a branch miss on this line.

Need to think about how to cover it

Reply:
This one is going to be hard to cover since the race window is very small here

Reply:
Its is truely unreachable now since `self._closed = True` is set right away and there is no race there anymore